git/working.go

```diff
@@ -169,22 +128,42 @@ func (c *Checkout) GetNote(ctx context.Context, rev string, note interface{}) (b
 	return getNote(ctx, c.dir, c.realNotesRef, rev, note)
 }
 
+// HeadRevision returns the revision of the current git HEAD
 func (c *Checkout) HeadRevision(ctx context.Context) (string, error) {
 	return refRevision(ctx, c.dir, "HEAD")
 }
 
-func (c *Checkout) SyncRevision(ctx context.Context) (string, error) {
-	return refRevision(ctx, c.dir, c.config.SyncTag)
+// GetSyncMarkerRevision returns the revision of the SyncMarker
+func (c *Checkout) GetSyncMarkerRevision(ctx context.Context) (string, error) {
```

Comment:
I'll give this a go and report back.  I believe it will hit the same problem I described above about how I get `SyncStatus` to have access to the right information to determine the state mode, but I'll give it a go and see what I get.

Reply:
I have updated the PR with an implementation of a `SyncProvider` (as with anything, name pending, but that's what it's called right now) that completely separates the implementation of the state storage for both the git sync tag and the "native" mode.